azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/ConvertUtils.java

```diff
@@ -0,0 +1,187 @@
+        final WebContainer webContainer = StringUtils.equalsIgnoreCase(webAppBase.javaContainer(), "java") ? WebContainer.JAVA_SE :
+                WebContainer.values().stream()
+                        .filter(container -> StringUtils.equals(javaContainer, container.getValue()))
+                        .findFirst().orElse(null);
+        return Runtime.getRuntime(OperatingSystem.WINDOWS, webContainer, javaVersion);
+    }
+
+    static RuntimeStack convertRuntimeToRuntimeStack(Runtime runtime) {
+        if (runtime.getOperatingSystem() != OperatingSystem.LINUX || runtime.getWebContainer() == null || runtime.getJavaVersion() == null) {
+            return null;
+        }
+        final String fixedLinuxJavaVersion = runtime.getJavaVersion().getValue().startsWith(JavaVersion.JAVA_8.getValue()) ? "8" : "11";
+        return RuntimeStack.getAll().stream().filter(runtimeStack -> {
+            final String[] versionArray = runtimeStack.version().split("-");
+            final String runtimeStackJavaVersion = versionArray[1];
+            final String runtimeStackContainer = StringUtils.join(runtimeStack.stack(), " ", versionArray[0]);
+            return StringUtils.containsIgnoreCase(runtimeStackJavaVersion, fixedLinuxJavaVersion) &&
+                    ((StringUtils.containsIgnoreCase(runtimeStackContainer, "java") && runtime.getWebContainer() == WebContainer.JAVA_SE) ||
+                            StringUtils.equalsIgnoreCase(runtimeStackContainer, runtime.getWebContainer().getValue()));
+        }).findFirst().orElse(null);
+    }
+
+    static com.azure.resourcemanager.appservice.models.WebContainer convertRuntimeToWebContainer(Runtime runtime) {
+        if (runtime.getOperatingSystem() != OperatingSystem.WINDOWS || runtime.getWebContainer() == null) {
+            return null;
+        }
+        if (runtime.getWebContainer() == WebContainer.JAVA_SE) {
+            return StringUtils.startsWith(runtime.getJavaVersion().getValue(), JavaVersion.JAVA_8.getValue()) ?
+                    com.azure.resourcemanager.appservice.models.WebContainer.JAVA_8 :
+                    com.azure.resourcemanager.appservice.models.WebContainer.fromString("java 11");
+        }
+        return com.azure.resourcemanager.appservice.models.WebContainer.values().stream()
+                .filter(container -> StringUtils.equalsIgnoreCase(container.toString(), runtime.getWebContainer().getValue()))
+                .findFirst().orElse(null);
+    }
+
+    static com.azure.resourcemanager.appservice.models.JavaVersion convertToServiceJavaVersionModel(Runtime runtime) {
+        if (runtime.getOperatingSystem() != OperatingSystem.WINDOWS || runtime.getJavaVersion() == null) {
+            return null;
+        }
+        return com.azure.resourcemanager.appservice.models.JavaVersion.values().stream()
+                .filter(serviceVersion -> StringUtils.equals(serviceVersion.toString(), runtime.getJavaVersion().getValue()))
```

Comment:
JavaVersion.values().stream()
                .filter(serviceVersion -> StringUtils.equals(serviceVersion.toString(), runtime.getJavaVersion().getValue()))
                .findFirst().orElse(null);

Reply:
Will fix in later PR.